### DISCUSSION TITLE
Nested collections, navigation, and content queries... Oh, my!

### DISCUSSION THREAD

**Discussion by AimForNaN** [statamic/cms#6010] [Q&A]
I'm looking for a way to form a URL pattern that matches my file structure. Assuming automatic handling of nested collections is supported (by means of collections), how would I go about in doing that? For example, I would want an entry at `/parent-collection/sub-collection/entry-slug.md` to have an automatically-generated route of `/parent-collection/sub-collection/entry-slug`.

If a tree-like structure is only possible through [Navigations](https://statamic.dev/navigation), then what is the facade equivalent to [templating](https://statamic.dev/navigation#templating)? Yet, even if automatic handling of nested collections is supported, I would still like to know the facade equivalent. How do I query for a navigation?

**Comment by afonic** [comment]
I don't think nested collections are supported. There are some workarounds I can think of, like using a single structured collection tree for all the items or creating a custom Laravel routes. If you explain your use case folks might suggest better ways though.

If I missed your point you can have a look at all the Facades here: https://github.com/statamic/cms/tree/3.3/src/Facades

**Reply by AimForNaN** [reply]
I had experimented with both [navigations](https://statamic.dev/navigation) and [ordered trees](https://statamic.dev/collections#ordering). However, there appears to be caveats for both. I can see why nested collections aren't a thing, because each collection represents an entire directory tree (infinitely deep; depth managed by collection), rather than a collection representing a single directory (one level deep). This approach is perfectly fine, so long as it doesn't translate to one route level.

For navigation, I managed to figure out I can access it via `Statamic::tag('nav:slug')`. This resolves structure, but not routing. (But I don't know how to access `Statamic::tag` outside a blade template.) Also, the returned structure eventually reduces down to arrays, which are much more cumbersome to work with. (But at least I can loop through arrays just fine.)

For ordered trees, I can't for the life of me get an actual reference to the tree of any given page. I've gotten close to what I want with ordered trees and root entries with the following route pattern `'/blog/{parent_uri}/{slug}'`, but I keep running into roadblocks.

What I am trying to achieve is the route matching the file structure no matter the depth. 
```
/blog/slug.md => /blog/slug
/blog/sub-folder/slug.md => /blog/sub-folder/slug
etc, etc...
```

Ordered trees have been the closest to getting what I want, but if I can't even figure out how to access the tree, I don't know what other options are there.

**Comment by afonic** [comment]
Generally, routing is managed by the collection. So the Statamic way to do what you want is a structured collection as you already figured out. However I don't think you can add files in sub directories and expect it to work by default, so unless someone can shed some light on which files you might need to override in order for it to keep working.

If you are able to keep them in one directory you can check NavTree and CollectionTree repositories in order to access the tree or modify it. See how the CP does it in the controller: https://github.com/statamic/cms/blob/3.3/src/Http/Controllers/CP/Collections/CollectionTreeController.php
